packages/editor/src/components/SavedBrowser.tsx

```diff
@@ -1,45 +1,176 @@
-import { useRecoilValue } from "recoil";
-import { localFilesState, workspaceMetadataSelector } from "../state/atoms.js";
+import { debounce } from "lodash";
+import { useCallback, useEffect, useRef } from "react";
+import { useRecoilCallback, useRecoilState, useRecoilValue } from "recoil";
+import { v4 as uuid } from "uuid";
 import {
-  useDeleteLocalFile,
-  useDuplicate,
+  autosaveTimerState,
+  currentAppUser,
+  currentWorkspaceState,
+  savedFilesState,
+} from "../state/atoms.js";
+import {
+  useDeleteWorkspace,
   useLoadLocalWorkspace,
-  useSaveLocally,
+  useSaveNewWorkspace,
+  useSaveWorkspace,
 } from "../state/callbacks.js";
+import { logInWrapper } from "../utils/firebaseUtils.js";
 import BlueButton from "./BlueButton.js";
 import FileButton from "./FileButton.js";
 
+/**
+ * Allows user to save workspace with ctrl+s or cmd+s
+ * Due to current layout scheme, this runs no matter what side tab the user
+ * is on. If layout scheme ever changes, this may need to be moved to App
+ */
+const saveShortcutHook = () => {
+  const saveWorkspace = useSaveWorkspace();
+
+  // Need useRecoilCallback for snapshot, otherwise currentWorkspace outdated
+  const handleShortcut = useRecoilCallback(
+    ({ snapshot }) =>
+      async ({
+        repeat,
+        metaKey,
+        ctrlKey,
+        key,
+      }: {
+        repeat: boolean;
+        metaKey: boolean;
+        ctrlKey: boolean;
+        key: string;
+      }) => {
+        const currentWorkspace = snapshot.getLoadable(
+          currentWorkspaceState,
+        ).contents;
+        if (repeat) return;
+        // Cmd+s or Cntrl+s
+        if (
+          (metaKey || ctrlKey) &&
+          key === "s" &&
+          currentWorkspace.metadata.location.kind == "stored" &&
+          !currentWorkspace.metadata.location.saved
+        ) {
+          saveWorkspace();
+        }
+      },
+  );
+
+  useEffect(() => {
+    document.addEventListener("keydown", handleShortcut, { passive: false });
+
+    // Cleanup
+    return () => document.removeEventListener("keydown", handleShortcut);
+  }, []);
+};
+
+/**
+ * Autosaves every 5 seconds after a user has finished editing
+ * Known bug: The first edit to a diagram after load will not trigger this.
+ * This is due to nuances with currentWorkspace state not updating in time.
+ * Due to current layout scheme, this runs no matter what side tab the user
+ * is on. If layout scheme ever changes, this may need to be moved to App
+ */
+const autosaveHook = () => {
```

Comment:
I repro'ed another case when autosave doesn't run:
* Load the IDE with wide layout (default left tab === "examples")
* Do not touch "saved" and just start editing
* No autosave (maybe expected)
* Manual save by Cmd+S
* Wait... and no autosave triggered

My theory is that `flexlayout` actually doesn't mount the component until it's expanded. Therefore, if the user never touches the tab, the hooks would never fire. 

My suggestion is to move the hook calls to `App` instead.